liblocation/src/main/java/com/mapbox/android/core/location/LocationPendingIntentProvider.java

```diff
@@ -0,0 +1,34 @@
+package com.mapbox.android.core.location;
+
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+class LocationPendingIntentProvider {
+  private static ArrayList<Integer> requestCodes = new ArrayList<>();
+
+  static LocationPendingIntent buildIntent(Context context, String action) {
+    Intent intent = new Intent(action);
+
+    PendingIntent broadcastPendingIntent = PendingIntent.getBroadcast(context, generateRequestCode(), intent, 0);
```

Comment:
The `requestCodes` list is never updated with the generated code so the `duplicateCode` check is not behaving as expected. Also, to make this solution clean, the `requestCodes` list should be updated whenever we request/remove location updates.